fca_import_scripts/lib/ext_to_sql.rb

```diff
@@ -0,0 +1,133 @@
+require 'csv'
+require 'date'
+require 'yaml'
+
+class ExtToSql
+  TIMESTAMP = DateTime.now.strftime('%Y-%m-%d %H:%M:%S.%N')
+  REPAIR_FILE = File.join(File.dirname(__FILE__), '..', 'repairs.yml')
+
+  def initialize(stderr = nil)
+    @stderr = stderr
+  end
+
+  def process_ext_file(path, &block)
+    File.foreach(path) do |line|
+      line = repair_line line
+      row = line.split('|')
+
+      next if row[0] == 'Footer'
+
+      if row[0] == 'Header'
+        @type = determine_type_from_header(row)
+        block.call build_copy_statement_for_type(@type)
+        log "  • \033[33;36mConverting #{@type} EXT to SQL.\033[0m ", newline: false
+        next
+      end
+
+      if record_active?(@type, row)
+        block.call build_row(row)
+      end
+
+      write_progress
+    end
+
+    block.call end_copy_statement
+  end
+
+  private
+
+  def determine_type_from_header(row)
+    case row[1]
+    when 'Alternative Firm Name'
+      return :subsidiary
+    when 'Firm Authorisation'
+      return :firm
+    when 'Individual Details'
+      return :adviser
+    else
+      fail "Unable to determine file type from header: #{row[1]}"
+    end
+  end
+
+  def build_copy_statement_for_type(type)
+    case type
+    when :adviser
+      'COPY lookup_advisers (reference_number, name, created_at, updated_at) FROM stdin;'
+    when :firm
+      'COPY lookup_firms (fca_number, registered_name, created_at, updated_at) FROM stdin;'
+    when :subsidiary
+      'COPY lookup_subsidiaries (fca_number, name, created_at, updated_at) FROM stdin;'
+    else
+      fail
+    end
+  end
+
+  def build_row(row)
+    number = row[0] # reference number for advisers or firms
```

Comment:
maybe named variables/constants for the different row numbers instead of comments?
row[0], row[1], row[4], row[19]